pkg/scheduler/framework/plugins/topologyspreadconstraints/utils_test.go

```diff
@@ -949,3 +950,1083 @@ func TestWillViolate(t *testing.T) {
+			clusters: []fleetv1beta1.MemberCluster{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: clusterName1,
+						Labels: map[string]string{
+							topologyKey1: topologyValue1,
+							topologyKey2: topologyValue1,
+							topologyKey3: topologyValue1,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: clusterName2,
+						Labels: map[string]string{
+							topologyKey1: topologyValue2,
+							topologyKey2: topologyValue1,
+							topologyKey3: topologyValue2,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: clusterName3,
+						Labels: map[string]string{
+							topologyKey1: topologyValue2,
+							topologyKey2: topologyValue2,
+							topologyKey3: topologyValue2,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: clusterName4,
+						Labels: map[string]string{
+							topologyKey1: topologyValue2,
+							topologyKey2: topologyValue2,
+							topologyKey3: topologyValue1,
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: clusterName5,
+						Labels: map[string]string{
+							topologyKey3: topologyValue1,
+						},
+					},
+				},
+			},
+			bindings: []*fleetv1beta1.ClusterResourceBinding{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: bindingName1,
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						TargetCluster: clusterName1,
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: bindingName2,
+					},
+					Spec: fleetv1beta1.ResourceBindingSpec{
+						TargetCluster: clusterName5,
+					},
+				},
+			},
+			doNotSchedule: []*fleetv1beta1.TopologySpreadConstraint{
+				{
+					MaxSkew:           &maxSkew2,
+					TopologyKey:       topologyKey1,
+					WhenUnsatisfiable: fleetv1beta1.DoNotSchedule,
+				},
+			},
+			scheduleAnyway: []*fleetv1beta1.TopologySpreadConstraint{
+				{
+					MaxSkew:           &maxSkew2,
+					TopologyKey:       topologyKey2,
+					WhenUnsatisfiable: fleetv1beta1.ScheduleAnyway,
+				},
+				{
+					MaxSkew:           &maxSkew1,
+					TopologyKey:       topologyKey3,
+					WhenUnsatisfiable: fleetv1beta1.ScheduleAnyway,
+				},
+			},
+			wantViolations: doNotScheduleViolations{
+				clusterName1: violationReasons{
+					fmt.Sprintf(doNotScheduleConstraintViolationReasonTemplate, topologyKey1, maxSkew2),
+				},
+			},
+			wantScores: topologySpreadScores{
+				clusterName2: -2*skewChangeScoreFactor - maxSkewViolationPenality,
+				clusterName3: -3 * skewChangeScoreFactor,
+				clusterName4: -2*skewChangeScoreFactor - maxSkewViolationPenality,
+				clusterName5: -maxSkewViolationPenality,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			state := framework.NewCycleState(tc.clusters, tc.bindings)
+
+			violations, scores, err := evaluateAllConstraints(state, tc.doNotSchedule, tc.scheduleAnyway)
+			if err != nil {
+				t.Errorf("evaluateAllConstraints() = %v, want no error", err)
```

Comment:
```suggestion
				t.Fatalf("evaluateAllConstraints() = %v, want no error", err)
```

not need to compare the result?

Reply:
Done.